addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/pii_disclosure.js

```diff
@@ -0,0 +1,92 @@
+
+// This script scan for the presence of Personally Information Identifiable in incoming WebSocket Messages.
+// More Specific, it passively scan messages for credit card numbers
+
+// * This script is based on org.zaproxy.zap.extension.pscanrulesAlpha.Piicanner
+
+// * Regex: https://regex101.com/r/RBY77J/3
+
+// Author: Manos Kirtas (manolis.kirt@gmail.com)
+
+OPCODE_TEXT = 0x1;
+RISK_HIGH 	= 3;
+CONFIDENCE_HIGH = 3;
+
+SEQUENCE_NUM = 3;
+
+creditCards = {
+    'American Express' : /\b(?:3[47][0-9]{13})\b/gm,
+    'Diners Club' :  /\b(?:3(?:0[0-5]|[68][0-9])[0-9]{11})\b/gm,
+    'Discover' : /\b(?:6(?:011|5[0-9]{2})(?:[0-9]{12}))\b/gm,
+    'Jcb' : /\b(?:(?:2131|1800|35\d{3})\d{11})\b/gm,
+    'Mestro' : /\b(?:(?:5[0678]\d\d|6304|6390|67\d\d)\d{8,15})\b/gm,
+    'Master Card' : /\b(?:(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12})\b/gm,
+    'Visa' : /\b(?:4[0-9]{12})(?:[0-9]{3})?\b/gm
+};
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    var counter = 0;
+    var message = String(msg.getReadablePayload()).valueOf();
+
+    var numberSequences = getNumberOfSequence(message,SEQUENCE_NUM);
+
+    numberSequences.forEach(function(sequence){
+        Object.keys(creditCards).forEach(function(creditCardType){
+
+            if((matches = sequence.match(creditCards[creditCardType])) != null){
+                matches.forEach(function(match){
+                    if(validateLuhnCheckSum(match)){
+
+                        helper.newAlert()
+                            .setRiskConfidence(RISK_HIGH, CONFIDENCE_HIGH)
+                            .setName("Personally Identifiable Information via WebSocket (script)")
+                            .setDescription("The response contains Personally Identifiable Information,"
+                                            + " such as CC number, SSN and similar sensitive data."
+                                            + "Credit Card type detected: " + creditCardType + ".")
+                            .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in cc7091dbd